firefox-ios/Client/Frontend/Home/HomepageViewController.swift

```diff
@@ -381,6 +385,15 @@ class HomepageViewController:
                                                          theme: theme)
         }
 
+        // Only dispatch action when user is in edit mode to avoid having the toolbar re-displayed
+        if featureFlags.isFeatureEnabled(.toolbarRefactor, checking: .buildOnly),
+           let toolbarState = store.state.screenState(ToolbarState.self, for: .toolbar, window: windowUUID),
+           toolbarState.addressToolbar.isEditing {
+            let action = ToolbarMiddlewareAction(windowUUID: windowUUID,
+                                                 actionType: ToolbarMiddlewareActionType.cancelEdit)
+            store.dispatch(action)
+        }
```

Comment:
Probably an edge case, but something I just thought of. Would it make sense to move this functionality to `scrollViewWillBeginDragging(_ scrollView: UIScrollView)` so that the user can search while the view is still scrolling?

Reply:
Yes, it would make sense to move it there but at the same time I am not sure it is really a use case since scrolling is something that is not very long in during usually. Definitely an edge case. For now I'd suggest to leave it like this.